### DISCUSSION TITLE
无法解析项目的父级 POM 文件 "top.tangyh.lamp:lamp-dependencies-parent:4.14.0"

### DISCUSSION THREAD

**Discussion by nizpxox** [zuihou/zuihou-admin-cloud#318] [01.常见问题]
"C:\Program Files\Java\jdk1.8.0_91\bin\java.exe" -Dmaven.multiModuleProjectDirectory=D:\Software\lamp-cloud-pro-datasource-column -Djansi.passthrough=true -Dmaven.home=D:\development\apache-maven-3.9.3 -Dclassworlds.conf=D:\development\apache-maven-3.9.3\bin\m2.conf "-Dmaven.ext.class.path=D:\IDEA2023\idea\IntelliJ IDEA 2023.1.2\plugins\maven\lib\maven-event-listener.jar" "-javaagent:D:\IDEA2023\idea\IntelliJ IDEA 2023.1.2\lib\idea_rt.jar=62064:D:\IDEA2023\idea\IntelliJ IDEA 2023.1.2\bin" -Dfile.encoding=UTF-8 -classpath D:\development\apache-maven-3.9.3\boot\plexus-classworlds-2.7.0.jar;D:\development\apache-maven-3.9.3\boot\plexus-classworlds.license org.codehaus.classworlds.Launcher -Didea.version=2023.1.2 -s D:\development\apache-maven-3.9.3\conf\settings.xml -Dmaven.repo.local=D:\m2\repository -DskipTests=true clean
[INFO] Scanning for projects...
Downloading from alimaven: http://maven.aliyun.com/nexus/content/groups/public/top/tangyh/basic/lamp-parent/4.14.0/lamp-parent-4.14.0.pom
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): Could not find artifact top.tangyh.basic:lamp-parent:pom:4.14.0 in alimaven (http://maven.aliyun.com/nexus/content/groups/public/) and 'parent.relativePath' points at no local POM @ line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
[FATAL] Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13
 @ 
[ERROR] The build could not read 12 projects -> [Help 1]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-dependencies-parent:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): Could not find artifact top.tangyh.basic:lamp-parent:pom:4.14.0 in alimaven (http://maven.aliyun.com/nexus/content/groups/public/) and 'parent.relativePath' points at no local POM @ line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-public:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-public\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-base:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-base\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-oauth:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-oauth\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-system:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-system\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-gateway:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-gateway\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-support:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-support\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-generator:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-generator\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-activiti:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-activiti\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-child:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-child\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-production:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-production\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR]   
[ERROR]   The project top.tangyh.lamp:lamp-sscommon:4.14.0 (D:\Software\lamp-cloud-pro-datasource-column\lamp-sscommon\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for top.tangyh.lamp:lamp-dependencies-parent:4.14.0: The following artifacts could not be resolved: top.tangyh.basic:lamp-parent:pom:4.14.0 (absent): top.tangyh.basic:lamp-parent:pom:4.14.0 was not found in http://maven.aliyun.com/nexus/content/groups/public/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of alimaven has elapsed or updates are forced and 'parent.relativePath' points at no local POM @ top.tangyh.lamp:lamp-dependencies-parent:4.14.0, D:\Software\lamp-cloud-pro-datasource-column\lamp-dependencies-parent\pom.xml, line 6, column 13 -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/UnresolvableModelException

Process finished with exit code 1

依赖项 "top.tangyh.basic:lamp-parent:pom:4.14.0"Maven 正在尝试从 http://maven.aliyun.com/nexus/content/groups/public/ 下载这些依赖项，但它似乎无法找到它们。

**Comment by zuihou** [comment]
看着文档来。 https://www.kancloud.cn/zuihou/zuihou-admin-cloud/2405108


先编译： https://github.com/zuihou/lamp-util
在编译 lamp-cloud

**Comment by nizpxox** [comment]
项目环境问题，已解决
